Hardcode region to 'us-east-1' for making bucket api

I am not sure that is a bug or intentional by hardcoding region to 'us-east-1' in making bucket api.
https://github.com/minio/minio-py/blob/188dba3dff087ef0d13e4d6336dcfcfc7c914f49/minio/api.py#L267

```Python
minio = Minio(endpoint=endpoint,
                            region='cn-north-1',
                            access_key=config['MINIO_ACCESS_KEY'],
                            secret_key=config['MINIO_SECRET_KEY'],
                            secure=False)

try:
    minio.make_bucket("test-bucket", location='cn-north-1')
except BucketAlreadyOwnedByYou:
    pass
```

I got the following error after running above code against Minio with MINIO_REGION=cn-north-1

`[2018-01-30T15:34:06.345832Z] [ERROR] {"method":"PUT","reqURI":"/test-bucket/","header":{"Accept-Encoding":["identity"],"Authorization":["AWS4-HMAC-SHA256 Credential=YIEVIPFF3GDAS5ET00ZL/20180130/us-east-1/s3/aws4_request, SignedHeaders=content-length;content-md5;host;user-agent;x-amz-content-sha256;x-amz-date, Signature=ed0ea1adfd52c84c5d984b712c82cbeaed81f4c49e68a9953070b001ade5e341"],"Content-Length":["154"],"Content-Md5":["QmSbD/Gu5Kkc7J7pdLTD/g=="],"Host":["localhost:9000"],"User-Agent":["Minio (Darwin; x86_64) minio-py/3.0.0"],"X-Amz-Content-Sha256":["281b1de7767d0eecc157ab442f0dc0befd88bcfbbff05a2be65adfd5a50457ea"],"X-Amz-Date":["20180130T153406Z"]}} (Signature Does Not Match)
`

--- a/minio/api.py
+++ b/minio/api.py
@@ -264,7 +264,7 @@ class Minio(object):
         url = self._endpoint_url + '/' + bucket_name + '/'
 
         # Get signature headers if any.
-        headers = sign_v4(method, url, 'us-east-1',
+        headers = sign_v4(method, url, location,
                           headers, self._access_key,
                           self._secret_key, content_sha256_hex)
 
@@ -296,6 +296,9 @@ class Minio(object):
 
         # default for all requests.
         region = 'us-east-1'
+        # region is set then use the region.
+        if self._region:
+            region = self._region
 
         # Get signature headers if any.
         headers = sign_v4(method, url, region,
@@ -1705,6 +1708,10 @@ class Minio(object):
         headers = {}
         # default for all requests.
         region = 'us-east-1'
+
+        # Region is set override.
+        if self._region:
+            return self._region
 
         # For anonymous requests no need to get bucket location.
         if self._access_key is None or self._secret_key is None:

